UM/TaskManagement/HttpRequestManager.py

```diff
@@ -0,0 +1,457 @@
+                                   timeout = timeout)
+
+    # Public API for creating an HTTP POST request. Returns a unique request ID for this request.
+    # Returns an HttpRequestData instance that represents this request.
+    def post(self, url: str,
+             headers_dict: Optional[Dict[str, str]] = None,
+             data: Optional[Union[bytes, bytearray]] = None,
+             callback: Optional[Callable[["QNetworkReply"], None]] = None,
+             error_callback: Optional[Callable[["QNetworkReply", "QNetworkReply.NetworkError"], None]] = None,
+             download_progress_callback: Optional[Callable[[int, int], None]] = None,
+             upload_progress_callback: Optional[Callable[[int, int], None]] = None,
+             timeout: Optional[float] = None) -> "HttpRequestData":
+        return self._createRequest("post", url, headers_dict = headers_dict, data = data,
+                                   callback = callback, error_callback = error_callback,
+                                   download_progress_callback = download_progress_callback,
+                                   upload_progress_callback = upload_progress_callback,
+                                   timeout = timeout)
+
+    # Public API for aborting a given HttpRequestData. If the request is not pending or in progress, nothing
+    # will be done.
+    def abortRequest(self, request: "HttpRequestData") -> None:
+        with self._request_lock:
+            # If the request is currently pending, just remove it from the pending queue.
+            if request in self._request_queue:
+                self._request_queue.remove(request)
+
+            # If the request is currently in progress, abort it.
+            if request in self._current_requests:
+                if request.reply is not None and request.reply.isRunning():
+                    request.reply.abort()
+                    Logger.log("d", "%s aborted", request)
+
+    # This function creates a HttpRequestData with the given data and puts it into the pending request queue.
+    # If no request processing call has been scheduled, it will schedule it too.
+    # Returns an HttpRequestData instance that represents this request.
+    def _createRequest(self, http_method: str, url: str,
+                       headers_dict: Optional[Dict[str, str]] = None,
+                       data: Optional[Union[bytes, bytearray]] = None,
+                       callback: Optional[Callable[["QNetworkReply"], None]] = None,
+                       error_callback: Optional[Callable[["QNetworkReply", "QNetworkReply.NetworkError"], None]] = None,
+                       download_progress_callback: Optional[Callable[[int, int], None]] = None,
+                       upload_progress_callback: Optional[Callable[[int, int], None]] = None,
+                       timeout: Optional[float] = None) -> "HttpRequestData":
+        assert(timeout is None or timeout > 0)
+        if timeout is not None and timeout <= 0:
```

Comment:
Although we don't do it, if we run Python in the optimized mode, which will generate `*.pyo` instead of `*.pyc`, certain commands will be ignored, such as `__debug__` and `assert()`. Basically I added the `if`-statement for the normal case, but we can remove the `assert`.